### DISCUSSION TITLE
Appium error

### DISCUSSION THREAD

**Discussion by SankaraNarayanan1609** [appium/appium#19948] [Q&A]
Hi everyone, I am new to appium, My code used to work but recently i am getting this error, need support ASAP.

C:\Users\SankaraNarayanan\.jdks\corretto-21.0.2\bin\java.exe -ea -Didea.test.cyclic.buffer.size=1048576 "-javaagent:C:\Users\SankaraNarayanan\AppData\Local\JetBrains\IntelliJ IDEA Community Edition 2023.3.6\lib\idea_rt.jar=53106:C:\Users\SankaraNarayanan\AppData\Local\JetBrains\IntelliJ IDEA Community Edition 2023.3.6\bin" -Dfile.encoding=UTF-8 -Dsun.stdout.encoding=UTF-8 -Dsun.stderr.encoding=UTF-8 -classpath "C:\Users\SankaraNarayanan\AppData\Local\JetBrains\IntelliJ IDEA Community Edition 2023.3.6\lib\idea_rt.jar;C:\Users\SankaraNarayanan\AppData\Local\JetBrains\IntelliJ IDEA Community Edition 2023.3.6\plugins\testng\lib\testng-rt.jar;C:\Users\SankaraNarayanan\IdeaProjects\QC\target\classes;C:\Users\SankaraNarayanan\.m2\repository\org\testng\testng\7.7.1\testng-7.7.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\slf4j\slf4j-api\1.7.36\slf4j-api-1.7.36.jar;C:\Users\SankaraNarayanan\.m2\repository\com\beust\jcommander\1.82\jcommander-1.82.jar;C:\Users\SankaraNarayanan\.m2\repository\org\webjars\jquery\3.6.1\jquery-3.6.1.jar;C:\Users\SankaraNarayanan\.m2\repository\io\appium\java-client\9.2.0\java-client-9.2.0.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-api\4.18.1\selenium-api-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-remote-driver\4.18.1\selenium-remote-driver-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\com\google\auto\service\auto-service-annotations\1.1.1\auto-service-annotations-1.1.1.jar;C:\Users\SankaraNarayanan\.m2\repository\com\google\guava\guava\33.0.0-jre\guava-33.0.0-jre.jar;C:\Users\SankaraNarayanan\.m2\repository\com\google\guava\failureaccess\1.0.2\failureaccess-1.0.2.jar;C:\Users\SankaraNarayanan\.m2\repository\com\google\guava\listenablefuture\9999.0-empty-to-avoid-conflict-with-guava\listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar;C:\Users\SankaraNarayanan\.m2\repository\com\google\code\findbugs\jsr305\3.0.2\jsr305-3.0.2.jar;C:\Users\SankaraNarayanan\.m2\repository\org\checkerframework\checker-qual\3.41.0\checker-qual-3.41.0.jar;C:\Users\SankaraNarayanan\.m2\repository\com\google\errorprone\error_prone_annotations\2.23.0\error_prone_annotations-2.23.0.jar;C:\Users\SankaraNarayanan\.m2\repository\com\google\j2objc\j2objc-annotations\2.8\j2objc-annotations-2.8.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\semconv\opentelemetry-semconv\1.23.1-alpha\opentelemetry-semconv-1.23.1-alpha.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-api\1.35.0\opentelemetry-api-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-context\1.35.0\opentelemetry-context-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-exporter-logging\1.35.0\opentelemetry-exporter-logging-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-sdk-common\1.35.0\opentelemetry-sdk-common-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-sdk-extension-autoconfigure-spi\1.35.0\opentelemetry-sdk-extension-autoconfigure-spi-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-sdk-extension-autoconfigure\1.35.0\opentelemetry-sdk-extension-autoconfigure-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-api-events\1.35.0-alpha\opentelemetry-api-events-1.35.0-alpha.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-sdk-trace\1.35.0\opentelemetry-sdk-trace-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-extension-incubator\1.35.0-alpha\opentelemetry-extension-incubator-1.35.0-alpha.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-sdk\1.35.0\opentelemetry-sdk-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-sdk-metrics\1.35.0\opentelemetry-sdk-metrics-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\io\opentelemetry\opentelemetry-sdk-logs\1.35.0\opentelemetry-sdk-logs-1.35.0.jar;C:\Users\SankaraNarayanan\.m2\repository\net\bytebuddy\byte-buddy\1.14.12\byte-buddy-1.14.12.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-http\4.18.1\selenium-http-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\dev\failsafe\failsafe\3.3.2\failsafe-3.3.2.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-json\4.18.1\selenium-json-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-manager\4.18.1\selenium-manager-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-os\4.18.1\selenium-os-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\apache\commons\commons-exec\1.3\commons-exec-1.3.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-support\4.18.1\selenium-support-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\com\google\code\gson\gson\2.10.1\gson-2.10.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-java\4.18.1\selenium-java-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-chrome-driver\4.18.1\selenium-chrome-driver-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-chromium-driver\4.18.1\selenium-chromium-driver-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-devtools-v120\4.18.1\selenium-devtools-v120-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-devtools-v121\4.18.1\selenium-devtools-v121-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-devtools-v122\4.18.1\selenium-devtools-v122-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-devtools-v85\4.18.1\selenium-devtools-v85-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-edge-driver\4.18.1\selenium-edge-driver-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-firefox-driver\4.18.1\selenium-firefox-driver-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-ie-driver\4.18.1\selenium-ie-driver-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\org\seleniumhq\selenium\selenium-safari-driver\4.18.1\selenium-safari-driver-4.18.1.jar;C:\Users\SankaraNarayanan\.m2\repository\ch\qos\logback\logback-classic\1.5.3\logback-classic-1.5.3.jar;C:\Users\SankaraNarayanan\.m2\repository\ch\qos\logback\logback-core\1.5.3\logback-core-1.5.3.jar" com.intellij.rt.testng.RemoteTestNGStarter -usedefaultlisteners false -socket53105 @w@C:\Users\SankaraNarayanan\AppData\Local\Temp\idea_working_dirs_testng.tmp -temp C:\Users\SankaraNarayanan\AppData\Local\Temp\idea_testng.tmp
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[Appium] Welcome to Appium v2.5.1
[Appium] The autodetected Appium home path: C:\Users\SankaraNarayanan\.appium
[Appium] Appium REST http interface listener started on http://0.0.0.0:4723
[Appium] You can provide the following URLs in your client code to connect to this server:
[Appium] 	http://192.168.100.102:4723/
[Appium] 	http://127.0.0.1:4723/ (only accessible from the same host)
[Appium] No drivers have been installed in C:\Users\SankaraNarayanan\.appium. Use the "appium driver" command to install the one(s) you want to use.
[Appium] No plugins have been installed. Use the "appium plugin" command to install the one(s) you want to use.
[HTTP] --> GET /status
[HTTP] {}
[AppiumDriver@9ee9] Calling AppiumDriver.getStatus() with args: []
[AppiumDriver@9ee9] Responding to client with driver.getStatus() result: {"ready":true,"message":"The server is ready to accept new connections","build":{"version":"2.5.1"}}
[HTTP] <-- GET /status 200 5 ms - 110
[HTTP] 
[HTTP] Request idempotency key: fc27fade-5ecb-4269-9ec4-4202c9e85b6d
[HTTP] --> POST /session
[HTTP] {"capabilities":{"firstMatch":[{}],"alwaysMatch":{"platformName":"ANDROID","appium:automationName":"UIAutomator2","appium:deviceName":"Pixel 8 Pro Android 15","appium:app":"C://Users//SankaraNarayanan//IdeaProjects//MobileApp//src//main//java//Resources//ApiDemos-debug.apk"}}}
[AppiumDriver@9ee9] Calling AppiumDriver.createSession() with args: [null,null,{"firstMatch":[{}],"alwaysMatch":{"platformName":"ANDROID","appium:automationName":"UIAutomator2","appium:deviceName":"Pixel 8 Pro Android 15","appium:app":"C://Users//SankaraNarayanan//IdeaProjects//MobileApp//src//main//java//Resources//ApiDemos-debug.apk"}}]
[AppiumDriver@9ee9] Event 'newSessionRequested' logged at 1711963460644 (14:54:20 GMT+0530 (India Standard Time))
[Appium] Attempting to find matching driver for automationName 'UIAutomator2' and platformName 'ANDROID'
[AppiumDriver@9ee9] Event 'newSessionStarted' logged at 1711963460650 (14:54:20 GMT+0530 (India Standard Time))
[AppiumDriver@9ee9] Encountered internal error running command: Error: Could not find a driver for automationName 'UIAutomator2' and platformName 'ANDROID'. Have you installed a driver that supports those capabilities? Run 'appium driver list --installed' to see. (Lower-level error: Could not find installed driver to support given caps)
[AppiumDriver@9ee9]     at DriverConfig.findMatchingDriver (C:\Users\SankaraNarayanan\AppData\Roaming\npm\node_modules\appium\lib\extension\driver-config.js:175:13)
[AppiumDriver@9ee9]     at AppiumDriver.createSession (C:\Users\SankaraNarayanan\AppData\Roaming\npm\node_modules\appium\lib\appium.js:644:35)
[AppiumDriver@9ee9]     at commandExecutor (C:\Users\SankaraNarayanan\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\basedriver\driver.ts:106:18)
[AppiumDriver@9ee9]     at AppiumDriver.executeCommand (C:\Users\SankaraNarayanan\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\basedriver\driver.ts:123:15)
[AppiumDriver@9ee9]     at processTicksAndRejections (node:internal/process/task_queues:95:5)
[AppiumDriver@9ee9]     at defaultBehavior (C:\Users\SankaraNarayanan\AppData\Roaming\npm\node_modules\appium\lib\appium.js:1105:16)
[AppiumDriver@9ee9]     at AppiumDriver.executeWrappedCommand (C:\Users\SankaraNarayanan\AppData\Roaming\npm\node_modules\appium\lib\appium.js:1215:16)
[AppiumDriver@9ee9]     at AppiumDriver.executeCommand (C:\Users\SankaraNarayanan\AppData\Roaming\npm\node_modules\appium\lib\appium.js:1121:17)
[AppiumDriver@9ee9]     at asyncHandler (C:\Users\SankaraNarayanan\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\protocol.js:393:19)
[HTTP] <-- POST /session 500 42 ms - 1122
[HTTP] 

org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: An unknown server-side error occurred while processing the command. Original error: Could not find a driver for automationName 'UIAutomator2' and platformName 'ANDROID'. Have you installed a driver that supports those capabilities? Run 'appium driver list --installed' to see. (Lower-level error: Could not find installed driver to support given caps) 
Host info: host: 'DESKTOP-1BMSTQL', ip: '192.168.100.102'
Build info: version: '4.18.1', revision: 'b1d3319b48'
System info: os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.2'
Driver info: io.appium.java_client.android.AndroidDriver
Command: [null, newSession {capabilities=[{appium:deviceName=Pixel 8 Pro Android 15, platformName=ANDROID, appium:automationName=UIAutomator2, appium:app=C://Users//SankaraNarayanan//IdeaProjects//MobileApp//src//main//java//Resources//ApiDemos-debug.apk}], desiredCapabilities=Capabilities {appium:app: C://Users//SankaraNarayanan..., appium:automationName: UIAutomator2, appium:deviceName: Pixel 8 Pro Android 15, platformName: ANDROID}}]
Capabilities {appium:app: C://Users//SankaraNarayanan..., appium:automationName: UIAutomator2, appium:deviceName: Pixel 8 Pro Android 15, platformName: ANDROID}

	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:139)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.appium.java_client.remote.AppiumProtocolHandshake.createSession(AppiumProtocolHandshake.java:133)
	at io.appium.java_client.remote.AppiumProtocolHandshake.createSession(AppiumProtocolHandshake.java:102)
	at io.appium.java_client.remote.AppiumCommandExecutor.createSession(AppiumCommandExecutor.java:176)
	at io.appium.java_client.remote.AppiumCommandExecutor.execute(AppiumCommandExecutor.java:237)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:519)
	at io.appium.java_client.AppiumDriver.startSession(AppiumDriver.java:268)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:162)
	at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:90)
	at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:102)
	at io.appium.java_client.android.AndroidDriver.<init>(AndroidDriver.java:109)
	at Appium.BaseScript.ConfigureAppium(BaseScript.java:30)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:69)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:361)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:296)
	at org.testng.internal.invokers.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:180)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:122)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.testng.TestRunner.privateRun(TestRunner.java:829)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:431)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:391)
	at org.testng.SuiteRunner.run(SuiteRunner.java:330)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
	at org.testng.TestNG.runSuites(TestNG.java:1099)
	at org.testng.TestNG.run(TestNG.java:1067)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:65)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:105)

Test ignored.

===============================================
Default Suite
Total tests run: 1, Passes: 0, Failures: 0, Skips: 1
Configuration Failures: 1, Skips: 1
===============================================

Process finished with exit code 0

**Comment by iRitul** [comment]
@SankaraNarayanan1609 : The error says :- "Could not find a driver for automationName 'UIAutomator2' and platformName 'ANDROID'. Have you installed a driver that supports those capabilities? Run 'appium driver list --installed' to see. "
Install uiautomator2 with command :-  **appium driver install uiautomator2** or using npm command.

**Reply by SankaraNarayanan1609** [reply]
Fixed, thank you.
